linkerd/detect/src/lib.rs

```diff
@@ -59,52 +61,67 @@ pub fn allow_timeout<P, T>((p, t): (DetectResult<P>, T)) -> (Option<P>, T) {
     }
 }
 
+// === impl Config ===
+
+impl<D: Default> Config<D> {
+    const DEFAULT_CAPACITY: usize = 1024;
+
+    pub fn from_timeout(timeout: time::Duration) -> Self {
```

Comment:
very minor nit: probably would have called this `with_timeout` since i would think of it as more of a 'constructor' than a 'conversion' :woman_shrugging: